### PR TITLE
Use runtime agnostic retry mechanism.

### PR DESCRIPTION
**Pull request by nihohit**
tokio_retry apparently doesn't work with async-std, and we didn't notice that because the tests enabled immediate reconnection. This happened because the reconnection was tested with a working server, so the connection manager could reconnect immediately.

### CODE REVIEW

**Comment by Xuanwo** [comment]
Hi, I'm so happy to discover that this PR already exists for Redis, just as I was about to add it! Thank you @nihohit for this.

**Review by Xuanwo — COMMENTED** [review]
Other looks good to me, thank you!

**Comment by nihohit** [comment]
Hi @Xuanwo happy to hear that this fixes an issue for you. Can you pull the branch and verify that this works for your setup?

File: redis/Cargo.toml
Comment (nihohit) [reply redis/Cargo.toml]: done

**Comment by nihohit** [comment]
@Xuanwo did you have a chance to test this?

**Comment by Xuanwo** [comment]
> @Xuanwo did you have a chance to test this?

Hi, I will find some time this weekend to give it a test.

**Comment by nihohit** [comment]
Thanks, waiting for your approval.

**Review by Xuanwo — APPROVED** [review]
Thank you @nihohit, this works well on my setup with redis-server.

**Review by jaymell — APPROVED** [review]

**Pull request merged**
